### PR TITLE
proposal to update CMakeLists.txt at top folder

### PR DESCRIPTION
**Pull request by wxmeng04**
Just suggestion.

### CODE REVIEW

**Comment by wxmeng04** [comment]
I prefer to use static lib. I'm not expecting to run several apps using the same SDL2 shared lib. So shared lib is not useful for me.

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,19 +3,25 @@ project(lvgl)
...
+set(CMAKE_C_FLAGS "-Wno-error=int-conversion")
 
 INCLUDE_DIRECTORIES(${PROJECT_SOURCE_DIR})
 
 file(GLOB_RECURSE INCLUDES "./*.h" )
 
+# Here is only an example to use static lib. If you use shared lib,
+# simply change SDL2-static to SDL2. Set SDL2_DIR to where SDL2Config.cmake
+# exists in the SDL2 package.
+set(SDL2_DIR "C:/Users/Administrator/Desktop/SDL/lib/lib/cmake/SDL2")
```
Comment (kisvegabor) [inline CMakeLists.txt:15]: I think we shouldn't hard code such a thing. Instead we can add to the README how to set `SDL2_DIR` when calling cmake.

**Review by kisvegabor — COMMENTED** [review]
Thank you. I prefer shared libs because on Linux they can be installed  with one command line instruction. 

If we'd go for static libs, maybe we can host an SDL version right in this repo and make both cmake and Eclipse use it. This way we can save the installation of SDL which can be cumbersome especially on Windows.

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,19 +3,25 @@ project(lvgl)
...
+find_package(SDL2 REQUIRED CONFIG REQUIRED COMPONENTS SDL2-static)
+include_directories(${SDL2_INCLUDE_DIRS})
+
 add_subdirectory(lvgl)
 
 SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/bin) 
-SET(CMAKE_CXX_FLAGS "-O3")
```
[the remark this answers is not in the document]
Comment (wxmeng04) [reply CMakeLists.txt:14]: You can maitain it definitely.
However I suppose the project use c compiler only. If it's the case CMAKE_CXX_FLAGS has no effect as it's a flag for c++ compiler.
Even if c++ compiler is used, I normally use the default optimization level. In the latest cmake version, it defaults to O3 as well. But I see some blog stating O3 is sometimes less efficient than O2 and they explicitly set O2 in CMAKE_CXX_FLAGS.

File: CMakeLists.txt (hunk 1 above)
Comment (wxmeng04) [reply CMakeLists.txt:15]: If you use an installer, SDL2 normally will be installed in some system folder. In this case SDL2 can be easily found by find_package() in module mode without setting SDL2_DIR.
However I prefer to compile the source code and install the SDL2 lib in some managed directory (It's a bad example here as I install the lib at the Desktop). In this case I have to specify SDL2_DIR explicitly before calling find_package(... CONFIG). If later I don't want to use SDL2, I delete the folder leaving my system clean.
It depends on your preference.

Comment (wxmeng04) [reply CMakeLists.txt:15]: If you use apt to install SDL2 for example, both shared and static lib will be installed. You can instruct CMake to use the specified lib by you. SDL2 means shared lib while SDL2-static means static lib.

**Comment by wxmeng04** [comment]
If it's clang16 complier, you must disable int-conversion error by `set (CMAKE_C_FLAGS "-Wno-error=int-conversion")`. I update my commit.

File: CMakeLists.txt (hunk 1 above)
Comment (kisvegabor) [reply CMakeLists.txt:15]: I think we should have a config which works out of the box for the most common scenarios, but make it open for customization. 


So although I'm not an experienced CMake user (so correct me if it doesn't work for any reason), I think `find_package()` would be better by default, but we should allow passing the path as a CMake parameter too.

File: CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (kisvegabor) [reply CMakeLists.txt:14]: I see, thanks.

**Pull request closed without merging**